owncloudApp/src/main/java/com/owncloud/android/workers/UploadFileFromContentUriWorker.kt

```diff
@@ -134,6 +147,34 @@ class UploadFileFromContentUriWorker(
         }
     }
 
+    private fun copyFileToLocalStorage() {
+        val inputStream = appContext.contentResolver.openInputStream(contentUri)
+
+        val localStorageProvider: LocalStorageProvider by inject()
+        cachePath = localStorageProvider.getTemporalPath(account.name) + uploadPath + UriUtils.getDisplayNameForUri(contentUri, appContext)
+        val cacheFile = File(cachePath)
+        val cacheDir = cacheFile.parentFile
+        if (cacheDir != null) {
+            if (!cacheDir.exists()) {
+                cacheDir.mkdirs()
+            }
+        }
+        cacheFile.createNewFile()
+
+        val outputStream = FileOutputStream(cachePath)
+        val buffer = ByteArray(4096)
+        if (inputStream != null) {
+            var count = inputStream.read(buffer)
+            while (count > 0) {
+                outputStream.write(buffer, 0, count)
+                count = inputStream.read(buffer)
+            }
+        }
```

Comment:
I think we could try to open the input and output streams when we are using them and we need to close them just after using them to release their resources.

```suggestion
        val inputStream = appContext.contentResolver.openInputStream(contentUri)
        val outputStream = FileOutputStream(cachePath)
        outputStream.use { fileOut ->
            inputStream?.copyTo(fileOut)
        }
        inputStream?.close()
```

We can take advantage of `FileOutputStream.copyTo `extension 👍

Reply:
Nice change!! Applied 👍